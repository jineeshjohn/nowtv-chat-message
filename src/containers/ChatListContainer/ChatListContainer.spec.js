import React from "react";
import { shallow } from "enzyme";
import { ChatListContainer } from "./index";

describe("<ChatListContainer>", () => {
  let wrapper = null;
  const chatMessage = [
    {
      messageId: "12356",
      userId: "613651251",
      fullName: "Robin Balmforth",
      timestamp: "2016-06-03T20:24:29Z",
      email: "robin@example.com",
      message: "Hello, World!",
      avatar: null
    }
  ];
  const fetchChatMessage = jest.fn();
  const props = {
    chatMessage,
    fetchChatMessage
  };

  beforeEach(() => {
    wrapper = shallow(<ChatListContainer {...props} />);
  });

  afterEach(() => {
    wrapper = null;
  });

  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have one record in the chatMessage", () => {
    expect(wrapper.props().chatMessage.length).toEqual(1);
  });

  it("should not have one record in the chatMessage", () => {
    expect(wrapper.props().chatMessage.length).toEqual(1);
  });
});
