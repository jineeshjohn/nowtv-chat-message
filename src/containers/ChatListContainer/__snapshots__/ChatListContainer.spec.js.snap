// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatListContainer> renders without crashing 1`] = `
<ChatList
  chatMessage={
    Array [
      Object {
        "avatar": null,
        "email": "robin@example.com",
        "fullName": "Robin Balmforth",
        "message": "Hello, World!",
        "messageId": "12356",
        "timestamp": "2016-06-03T20:24:29Z",
        "userId": "613651251",
      },
    ]
  }
  key="ChatList"
/>
`;
